src/node.cc

```diff
@@ -851,11 +878,24 @@ void PromiseRejectCallback(PromiseRejectMessage message) {
 
   if (event == v8::kPromiseRejectWithNoHandler) {
     callback = env->promise_reject_unhandled_function();
-    value = message.GetValue();
 
+    // Exit the process in case the unhandled promise is garbage collected
+    // without being handled later on.
+    Persistent<Promise>* p = new Persistent<Promise>(env->isolate(), promise);
+    p->SetWeak(p, WeakCallback, WeakCallbackType::kFinalizer);
+    CHECK_EQ(promise->GetAlignedPointerFromInternalField(1), nullptr);
+    promise->SetAlignedPointerInInternalField(node::kUnhandledPromiseField, p);
```

Comment:
No need to use the `node::` qualifier here and on lines 865, 985 and 898.  Use the constant on line 886.